src/Microsoft.ML.TimeSeries/PredictionEngine.cs

```diff
@@ -0,0 +1,207 @@
+﻿using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.ML.TimeSeries
+{
+    public delegate void ValuePinger(ref bool value, int rowPosition);
+
+    public interface IStatefulRow : IRow
+    {
+        ValuePinger[] GetPingers();
```

Comment:
>ValuePinger[] [](start = 8, length = 13)

1. Why do you need value (and why of type bool)? 
2. Why this is an array?

Reply:
I believe you can get away with just `Action<long> GetPinger`

---
In reply to: [235552903](https://github.com/dotnet/machinelearning/pull/1618#discussion_r235552903) [](ancestors = 235552903)

Reply:
Or maybe `void Ping(long position)`

---
In reply to: [235552952](https://github.com/dotnet/machinelearning/pull/1618#discussion_r235552952) [](ancestors = 235552952,235552903)

Reply:
Or `void Ping()`

---
In reply to: [235553822](https://github.com/dotnet/machinelearning/pull/1618#discussion_r235553822) [](ancestors = 235553822,235552952,235552903)